FirstConsoleApp/StringCalculator.cs

```diff
@@ -1,32 +1,60 @@
 using System.Net.Sockets;
 using System;
+using System.Linq;
 
 namespace FirstConsoleApp
 {
     public class StringCalculator
     {
-        public StringCalculator()
-        {
-        }
+        public StringCalculator() { }
 
         public int Add(string input)
         {
-            if (input == "") return 0;
-            char[] separators = {',' , '\n' };
-            string[] numbers = input.Split(separators);
+            if (!input.Any(Char.IsDigit)) return 0;
+
+            int[] numbers = GetNumbersArray(input);
             int sum = 0;
+
             foreach (var item in numbers)
             {
-                bool isNumber = int.TryParse(item, out int result);
-                if (!isNumber)
-                {
-                    return 0;
-                }
-
-                sum += result;
+                sum += item;
             }
             
             return sum;
         }
+
+        private int[] GetNumbersArray(string input)
+        {
+            string[] numbers;
+            bool isSeparatorSpecifiedinInput = input.StartsWith("//");
+            if (isSeparatorSpecifiedinInput)
+            {
+                string separator = GetSeparator(input);
+                var numbersList = GetNumbersList(input);
+                numbers = numbersList.Split(separator);
+            }
+            else
+            {
+                char[] defaultSeparators = {',', '\n'};
+                numbers = input.Split(defaultSeparators);
+            }
+
+            return Array.ConvertAll(numbers,int.Parse);
+        }
+
+        private string GetNumbersList(string input)
```

Comment:
Hmmm, think the naming of the functions `GetNumbersArray` and `GetNumbersList` can still be improved. 

`GetNumbersList` returns a string so it doesn't really represent its name. When I look at the function, what it's actually doing is returning a part of the string it's been passed in.

`GetNumbersArray` although returns an array, may not be an appropriate name still because if the implementation changes and it returns an `IEnumerable` or `List` instead, the name would need to be changed too. Maybe you can rename this to something more generic like `GetNumbers` or `GetNumbersFromInput`?